### ISSUE TITLE
RT2870USBWirelessDriver.kext isn't being loaded (Mojave 10.14.3)

### ISSUE THREAD

**Issue by anonym24** [chris1111/USB-Wireless-Utility#1]
Previously I used pgk to install wireless utility form my official USB WiFi manufacture (W311MAV3.0 Driver For MAC installer from https://tendacn.com/en/product/download/W311MA.html)
That installer's pkg could only install framework and GUI but it failed to install .kext
So I unzipped kext from pkg and copied manually, rebuilt kext cache, rebooted and everything was working ok.

Your version installs kext file into Extension folder ok but it is never loaded (`kextstat` doesn't show it)

I tried to rebuild cache, rebooted, but still it wasn't there

![image](https://user-images.githubusercontent.com/8851301/54494554-39254d00-48e4-11e9-83be-eb86254d0898.png)

And what version of RT2870USBWirelessDriver.kext does your pkg installs for Mojave OS?
Does this pkg contains the same old drivers (kext files) and there is really no point to use your version?

Previously I used kext file form Yosemite folder and it was working ok with Mojave until my first system update (I unzipped kext from pkg inside MT7610U_D5.0.1.25_SDK1.0.2.18_UI5.0.0.27_20150429.dmg)

![image](https://user-images.githubusercontent.com/8851301/54494620-f9129a00-48e4-11e9-9fe9-fe0f59317857.png)

After one update boot failed due to this kext

This user from tonymacx86 helped me to detect the problem with booting after update (that this issue was related to this kext driver file) https://www.tonymacx86.com/threads/os-x-10-11-4-update.187001/post-1224400 or https://www.insanelymac.com/forum/topic/309352-el-capitan-on-ga-z170x-gaming-5/?do=findComment&comment=2226050 

![image](https://user-images.githubusercontent.com/8851301/54494681-a1286300-48e5-11e9-834d-a8af3903e981.png)

**Comment by anonym24** [comment]
I see that your pkg contains a different kext for more newer operations systems (sierra, high sierra and mojave), 1.4 mb kext is used for these OS

![image](https://user-images.githubusercontent.com/8851301/54494994-bd79cf00-48e8-11e9-8dd1-94dce3ecb879.png)

Mb the installer didn't set the correct rights for `RT2870USBWirelessDriver.kext` file inside `Extensions` folder? I'll try to set it manually or mb install this kext using KextBeast 2.0.2

**Comment by anonym24** [comment]
actually my previous driver also stopped working

Utility won't start when I click on it

![image](https://user-images.githubusercontent.com/8851301/54495650-519a6500-48ee-11e9-9d50-b36562f217a0.png)

**Comment by anonym24** [comment]
I installed driver again, rebooted, then shutdown the PC. Unplugged WiFi USB stick and replugged it to a different USB port, turned on the PC and it started to working. I don't know how it works.. Now `kextstat` says that kext is loaded:

![image](https://user-images.githubusercontent.com/8851301/54495972-6a0c7e80-48f2-11e9-8ba7-d665483d1efa.png)

**Comment by chris1111** [comment]
This program is not for use with other programs and especially after trying many other things.
It works perfectly when used as it should.
issue is closed

**Comment by anonym24** [comment]
I didn't said that I used it with other programs. I said that I used some previous version from my manufacture and then deleted and tried this program.
Anyway, it seems you don't care what write here anyway.

I just need to know that macOS it's not Windows. It doesn't have have a great support of different hardware and other stuff. It doesn't even support Android MTP and Android USB Tethering until you install 3rd party software
